spec/requests/api/journalist_can_create_an_article_spec.rb

```diff
@@ -87,4 +95,47 @@
       expect(response_json['message']).to eq 'Category must exist'
     end
   end
+
+  describe 'Registered user can not create an article' do
+    before do
+      post '/api/articles',
+           params: {
+             article: {
+               title: 'A title',
+               lead: 'And a lead',
+               body: 'With a body',
+               category_id: category.id,
+               author_id: registered_user.id
+             }
+           },
+           headers: registered_user_headers
+    end
+    it 'is expected to return 401 status' do
+      expect(response).to have_http_status 401
+    end
+    it 'is expected to return an error message' do
+      expect(response_json).to have_key('message').and have_value('You are not authorized to create an article.')
+    end
+  end
+
+  describe 'Visitor can not create an article' do
+    before do
+      post '/api/articles',
+           params: {
+             article: {
+               title: 'A title',
+               lead: 'And a lead',
+               body: 'With a body',
+               category_id: category.id,
+               author_id: 'Test author'
+             }
+           }
+    end
+    it 'is expected to return 401 status' do
+      expect(response).to have_http_status 401
+    end
+    it 'is expected to return an error message' do
+      expect(response_json['errors']).to eq ['You need to sign in or sign up before continuing.']
```

Comment:
Not sure if this is the message you want to send, but more importantly, please stick to the same standards you use above in this expect statement. `have_key` and `have_value` are good...

Reply:
Good input, we have now corrected tests for better consistency!